libyul/ObjectParser.cpp

```diff
@@ -120,7 +120,10 @@ void ObjectParser::parseData(Object& _containingObject)
 
 	YulString name = parseUniqueName(&_containingObject);
 
-	expectToken(Token::StringLiteral, false);
+	if (currentToken() == Token::HexStringLiteral)
+		expectToken(Token::HexStringLiteral, false);
+	else
+		expectToken(Token::StringLiteral, false);
```

Comment:
@chriseth Is there a better way to do this? Maybe a function like `expectToken` that works for multiple tokens instead of a single one?